Translate (#7)
LittleSkinCommspt/LittleSkinManualDev: en/advanced/yggdrasil.md

# Yggdrasil

LittleSkin provide Yggdrasil Verify Authentication Service. You can use it with [authlib-injector](https://github.com/yushijinhun/authlib-injector) to load skin from LittleSkin without skin loader mod.

You can download the latest version of authlib-injector at <https://authlib-injector.yushi.moe/~download/>

## Yggdrasil API address

The Yggdrasil API address of LittleSkin is:

```
https://littlesk.in/api/yggdrasil
```

> LittleSkin has already active the API Address Indication (ALI) function of authlib-injector. When you are using launcher that supported authlib-injector of ALI, type any address of LittleSkin can be identified.

Reviewer: > 这里“启用”应该是 enable，active 有种让这个功能“活跃”起来、能够自己有意识地去处理某些东西的感觉（而且 active 的动词是 activate，“激活”），而 enable 就很简单地表示让其“可以这么做”
> 
> https://wikidiff.com/enable/activate

```suggestion
LittleSkin has enabled API Address Indication (ALI). When using authlib-injector and a Minecraft launchers with ALI support, all URL of LittleSkin could be used as API root.
```

基于 @tnqzh123 的建议的进一步改进，读起来更通顺

## API Documents

See: [Yggdrasil Server Technical Specifications](https://github.com/yushijinhun/authlib-injector/wiki/Yggdrasil%E6%9C%8D%E5%8A%A1%E7%AB%AF%E6%8A%80%E6%9C%AF%E8%A7%84%E8%8C%83)

## Use at server

Please change the `online-mode` value to `true` of server configuration file, and add this parameter before `-jar` of your server start command:

```
-javaagent:{path/to/authlib-injector.jar}=https://littlesk.in/api/yggdrasil
```

In this, `{path/to/authlib-injector.jar}` is the path that direct to authlib-injector.

::: warning

> <strong>The path that you write isn't has big parantheses in real!<strong>The big parantheses is only show that must be assign this value correctly, but needn't write big paratheses in real.

Reviewer: > **The path you set SHOULDN'T contain parentheses in the both side!** The parentheses is just meaning that you must set this value correctly, but it shouldn't appears in your path.

```suggestion
**The path you set SHOULDN'T contain parentheses on both sides!** The parentheses just means that you must set this value correctly, but it shouldn't appear in your path.
```

:::

::: tip

> If you're using BungeeCord, you need to load authlib-injector for all server(include BungeeCord and all sub servers), but you can only open `online-mode` of BungeeCord, keep closed for other server.

Reviewer: > keep it DISABLED in other servers.

keep it DISABLED **on** other servers. ？

:::

## Use at client

> Your launcher need to support custom Yggdrasil server if use Yggdrasil of LittleSkin in client. We are recommended to use [HMCL 3](https://www.mcbbs.net/thread-142335-1-1.html) and [BakaXL](https://www.mcbbs.net/thread-512144-1-1.html).

Reviewer 2: ```suggestion
Make sure your launcher supports using a custom Yggdrasil server before proceeding. It's always a great choice to take a look at [HMCL 3](https://github.com/huanghongxun/HMCL) or [BakaXL](https://www.mcbbs.net/thread-512144-1-1.html) when you can't decide.
```

::: tip

> If your launcher doesn't support custom Yggdrasil, and the launcher is written in Java, you can add JVM Arguments to load authlib-injector(same as use authlib-injector at server), but at this document we won't show how to do this.

Reviewer 2: ```suggestion
For launchers written in java, you can force loading authlib-injector by adding JVM arguments. Such method works regardless of launchers' support of using custom Yggdrasil API,  It's the same as doing that on a server and will not be described in detail here.
```

:::

> We use HMCL 3 to set up an example to show how to use Yggdrasil in Minecraft Client.

Reviewer 2: ```suggestion
Here we use HMCL 3 to demonstrate how to use LittleSkin's Yggdrasil API in a Minecraft client:
```

  
1. Open the "Create a new account" page, select the login type as "authlib-injector".
If this is your first time to open HMCL, it will appear when it started.

![set-login-method](./assets/yggdrasil/set-login-method.png)

> 2. Click the "+" at the right of auth server, type the Yggdrasil API address of LittleSkin, click "Next", wait for it identify the Yggdrasil of LittleSkin, then click "Finish".
> If you already add the Yggdrasil of LittleSkin you can select LittleSkin at auth server list.

Reviewer 2: ```suggestion
2. Click the "+" button at the right of "Authentication Server", enter the Yggdrasil API root of LittleSkin at the text field, and click "Next". Please wait a moment for the launcher to retrieve information from our website. Once finished, click "Finish".
If you have added our Yggdrasil API before, simply select LittleSkin from the authentication server list.
```

![add-yggdrasil-server-1](./assets/yggdrasil/add-yggdrasil-server-1.png)

![add-yggdrasil-server-2](./assets/yggdrasil/add-yggdrasil-server-2.png)

![add-yggdrasil-server-3](./assets/yggdrasil/add-yggdrasil-server-3.png)

3.Select LittleSkin at auth server list, then type you E-Mail and password. "Name" is the E-Mail address of your LittleSkin account.

> If you only add a auth server, so HMCL 3 will select the only auth server defaultly.

Reviewer 2: ```suggestion
3. Select LittleSkin from the authentication server list, and type your email address and password. "Name" is your email address bound to your LittleSkin account.
When only one authentication server is present, that server will be the default choice.
```

![set-email-password](./assets/yggdrasil/set-email-password.png)

> 4.Select your character.
> If you only have a character, so HMCL 3 will select the only character defaultly.
> If you wanna add more characters, you can add your account again to add other character.

Reviewer 2: ```suggestion
4. Select your player.
When there's only one player associated with the account, that player will be the default choice.
Repeat 1-4 until you have added all necessary players.
```

![choose-player](./assets/yggdrasil/choose-player.png)

> 5.Setup is complete! You can start game now.

Reviewer 2: ```suggestion
5. You're all set! It's time to enjoy a game with your favourite skin.
```

> ::: tip A point
> For PCL2, you might be need to configure Yggdrasil server address by yourself at `Version settings` → `Settings` → `Server options`.
> 
> Login type：`Authlib-Injector`
> Auth server：`https://mcskin.littleservice.cn/api/yggdrasil`
> Register address：`https://mcskin.littleservice.cn/auth/register`
> Server name：`LittleSkin`
> :::

Reviewer 2: ```suggestion
::: tip TIPS
For PCL 2(another Minecraft launcher), you might need to configure Yggdrasil server at Version settings → Settings → Server options with following parameters:
Login Type: Authlib-Injector
Auth Server: https://littleskin.cn/api/yggdrasil
Register Address: https://littleskin.cn/auth/register
Server Name：LittleSkin
:::
```
